scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,68 +117,160 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
-        DocumentId documentId = DocumentId.create(documentName, defaultModuleId);
-        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), documentName);
+        DocumentId documentId = DocumentId.create(pluginImportsDocumentName, defaultModuleId);
+        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(),
+                pluginImportsDocumentName);
         defaultModule.modify().addDocument(documentConfig).apply();
-
         project.currentPackage().ballerinaToml().ifPresent(ballerinaToml -> {
             String tomlFileContent = ballerinaToml.tomlDocument().textDocument().toString();
             ballerinaToml.modify().withContent(tomlFileContent + tomlDependencies).apply();
         });
 
-        List<ScannerContext> scannerContextList = new ArrayList<>();
+        PackageResolution packageResolution = project.currentPackage().getResolution();
+        ResolvedPackageDependency rootPkgNode = new ResolvedPackageDependency(project.currentPackage(),
+                PackageDependencyScope.DEFAULT);
+        List<Package> directDependencies = packageResolution.dependencyGraph().getDirectDependencies(rootPkgNode)
+                .stream().map(ResolvedPackageDependency::packageInstance).toList();
+        Map<String, List<Rule>> externalAnalyzers = new HashMap<>();
+        for (Package pkgDependency : directDependencies) {
+            PackageManifest pkgManifest = pkgDependency.manifest();
+            String org = pkgManifest.org().value();
+            String name = pkgManifest.name().value();
+            String pluginName = org + FORWARD_SLASH + name;
+            if (pkgManifest.compilerPluginDescriptor().isEmpty() || !analyzerDescriptors.contains(pluginName)) {
+                continue;
+            }
 
-        // TODO: Implement the logic to pass scanner contexts to compiler plugins and perform external static analysis.
+            CompilerPluginDescriptor pluginDesc = pkgManifest.compilerPluginDescriptor().get();
+            List<URL> jarUrls = pluginDesc.dependencies().stream().map(dependency -> {
+                        try {
+                            return Path.of(dependency.getPath()).toUri().toURL();
+                        } catch (MalformedURLException ex) {
+                            throw new RuntimeException(ex);
+                        }
+            }).toList();
 
-        List<Issue> externalIssues = new ArrayList<>(scannerContextList.size());
-        for (ScannerContext scannerContext : scannerContextList) {
-            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
-            externalIssues.addAll(reporter.getIssues());
+            URLClassLoader ucl = URLClassLoader.newInstance(jarUrls.toArray(URL[]::new),
+                    this.getClass().getClassLoader());
+            InputStream resourceAsStream = ucl.getResourceAsStream(RULES_FILE);
+            if (resourceAsStream == null) {
```

Comment:
Is this being null an acceptable possibility?

Reply:
Yes, If the `rules.json` file is not included in a compiler plugin, the outcome of `ucl.getResourceAsStream()` becomes `null`

Reply:
Yeah, my question is if that is acceptable? In that case shouldn't there be an error (or warning) somewhere? You're just continuing here atm?

Reply:
> Yeah, my question is if that is acceptable? In that case shouldn't there be an error (or warning) somewhere? You're just continuing here atm?

The suggestion is correct, will include an exception at this point.